packages/graph/hash_graph/lib/graph/src/types/mod.rs

```diff
@@ -1,50 +1,74 @@
-//! Descriptions, and implementation logic, of the Types of elements of the graph.
+//! Model types used across datastores
 
-use sqlx::{types::Uuid, FromRow};
+use std::fmt;
 
-#[derive(Debug, FromRow)]
-pub struct DataType {
-    pub version_id: Uuid,
-    pub schema: serde_json::Value,
-    pub created_by: Uuid,
+use uuid::Uuid;
+
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, sqlx::Type, PartialEq, Eq)]
+#[sqlx(transparent)]
+pub struct AccountId(Uuid);
+
+impl AccountId {
+    #[must_use]
+    pub const fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct PropertyType {
-    pub version_id: Uuid,
-    pub schema: serde_json::Value,
-    pub created_by: Uuid,
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, sqlx::Type, PartialEq, Eq)]
+#[sqlx(transparent)]
+pub struct BaseId(Uuid);
+
+impl BaseId {
+    #[must_use]
+    pub const fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct EntityType {
-    pub version_id: Uuid,
-    pub schema: serde_json::Value,
-    pub created_by: Uuid,
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, sqlx::Type, PartialEq, Eq)]
+#[sqlx(transparent)]
+pub struct VersionId(Uuid);
+
+impl VersionId {
+    #[must_use]
+    pub const fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
 }
 
-#[derive(Debug, FromRow)]
-pub struct Entity {
-    pub entity_id: Uuid,
-    pub source_entity_type_version_id: Uuid,
-    pub properties: serde_json::Value,
-    pub created_by: Uuid,
+#[derive(Clone, Debug)]
+pub struct Identifier {
+    pub base_id: BaseId,
+    pub version_id: VersionId,
 }
 
-#[derive(Debug, FromRow)]
-pub struct PropertyTypePropertyTypeReference {
-    pub source_property_type_version_id: Uuid,
-    pub target_property_type_version_id: Uuid,
+impl fmt::Display for Identifier {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // TODO: change this to not just he the debug impl.
```

Comment:
```suggestion
        // TODO: change this to not just using the debug impl.
```

Why'd we remove the version you guys had when we were on a call? I know I raised some issues with it but like I said, I thought it was fine for now :)

Reply:
Tim pointed out the output looked almost identical to the default debug print, so we opted for that instead (just after you left the call :sweat_smile: )